codec-http/src/test/java/io/netty/handler/codec/http/HttpRequestEncoderTest.java

```diff
@@ -154,4 +157,81 @@ public void testEmptydBufferShouldPassThrough() throws Exception {
         channel.finishAndReleaseAll();
         assertEquals(0, buffer.refCnt());
     }
+
+    @Test
+    public void testForChunkedRequestNpe() throws Exception {
+        EmbeddedChannel channel = new EmbeddedChannel(new HttpRequestEncoder());
+        channel.writeOutbound(new CustomHttpRequest());
+        channel.writeOutbound(new DefaultHttpContent(Unpooled.copiedBuffer("test", CharsetUtil.US_ASCII)));
+        channel.writeOutbound(LastHttpContent.EMPTY_LAST_CONTENT);
+    }
+
+    private static class CustomHttpRequest implements HttpRequest {
```

Comment:
why is this object necessary? can you just use `DefaultHttpRequest` or is class loading impacted? Can you add a comment if it is necessary?

Reply:
Yes. If DefaultHttpRequest is used, the HttpHeaders class will be initialized before HttpUtil and the test won't trigger the issue.

Reply:
final

Reply:
@mcnulty - Please add a java doc or comment including this rational.